### ISSUE TITLE
'Use Range' code fix does not work with single-argument 'Substring'

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#40439]
**Version Used**: fb028b7c198c2f726067bb8627cb014f1154a541

**Steps to Reproduce**:

Apply the code fix for IDE0057 (Use range) on the following code:

```
string text = "...";
string subText = text.Substring(1);
```

**Expected Behavior**:

```
string text = "...";
string subText = text[1..];
```

**Actual Behavior**:

No diagnostic is reported, so the code fix cannot be used.

**Comment by zaytsev-victor** [comment]
@sharwell 
What about this case?
```csharp
string text = "...";
string subText = text.Substring(1, 2);
```
Currently this refactoring is suggested only if method has 2 parameters and one of them is subtraction of length and any value.

**Comment by sharwell** [comment]
@zaytsev-victor I talked with @CyrusNajmabadi about that case, and we believe that the current function call syntax has an advantage if the length parameter is a fixed value (i.e. the end position is relative to the start, as opposed to being relative to the end of the string). This is why the refactoring for calls to `Substring(int, int)` is only enabled when the second argument involves subtraction.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/56296
